library/common-library/templates/_naming.tpl

```diff
@@ -18,13 +28,16 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 {{- define "newrelic.common.naming.fullname" -}}
 {{- $name := "" }}
 
-{{- if .Values.fullnameOverride }}
-    {{- $name = .Values.fullnameOverride  }}
-{{- else }}
-    {{- $name = printf "%s-%s" .Release.Name (include "newrelic.common.naming.name" .)}}
-{{- end }}
+{{- if .Values.fullnameOverride -}}
+    {{- $name = .Values.fullnameOverride  -}}
+{{- else -}}
+    {{- $name = include "newrelic.common.naming.name" .  -}}
+    {{- if not ( contains $name .Release.Name ) -}}
```

Comment:
could you update the func description with this logic?

Reply:
Solved with [#32cb46f](https://github.com/newrelic/helm-charts/pull/784/commits/32cb46f39c32b55e3fdf213bb82409c83d813ff2)